Automattic/woocommerce-payments
# Fix the menu badge inconsistency on wp.com

Fixes #3941 

#### Changes proposed in this Pull Request

This PR fixes the menu badge inconsistency on wp.com sites by using `<span class="wcpay-menu-badge awaiting-mod count-1"><span class="plugin-count">1</span></span>`

The current code uses `<span class="wcpay-menu-badge awaiting-mod count-1">1</span>';` to add (1) badge to the menu, but it doesn't work on calysofied pages such as https://wordpress.com/home/:site-url.wpcomstaging.com


#### Testing instructions

**Testing locally**

1. Start with a fresh install. Make sure you're not connected.
2. Update `wcpay_activation_timestamp` to at least 3 days in the past.

```
Update wp_options set optionvalue='1646161274' where option_name='wcpay_activation_timestamp'
```
3. Navigate to the admin area and confirm the Wc Payments menu has 1 count.

**Testing on a wp.com site**

1. Sign up for a business or ecom plan on wp.com.
2. Convert the site into a WoA dev blog and enable SSH access.
3. Install WooCommerce and WC Payments.
4. Update `wcpay_activation_timestamp` to at least 3 days in the past.

```
Update wp_options set optionvalue='1646161274' where option_name='wcpay_activation_timestamp'
```
5. SSH into the server and open `includes/class-wc-payments-admin.php`. Apply changes in this branch.
7. Navigate to `https://wordpress.com/home/:site-url.wpcomstaging.com` and confirm the fix.
